### DISCUSSION TITLE
Which IDE or developing tools do the elementary os's team use?

### DISCUSSION THREAD

**Discussion by sauntor** [elementary/triage#274] [Q&A]
### What Happened?

I want to program some features to some apps, but I find it's hard to get a comfortable developing environment, can you show what tools you developing with?

It's frustrating for me to code as editing normal plain text...

### Steps to Reproduce

N/A.

### Expected Behavior

Maybe you can give the configs or plugins to installing for some common used tools like `VSCode`   or `VIM`

### OS Version

7.x (Horus)

### Software Version

Latest release (I have run all updates)

**Comment by jeremypw** [comment]
I use the native elementary apps - Code (for editing) and Terminal (for building and pushing/pulling) along with `io.elementary.vala-lint` and `git-gui`.  I check on https://vala-doc.org for syntax if necessary. There may be more powerful IDEs but we should "eat our own dogfood" :smile:

**Comment by lenemter** [comment]
I use Visual Studio Code for editing and most times as terminal while developing, sometimes use built-in Terminal app. I use [Vala plugin](https://marketplace.visualstudio.com/items?itemName=prince781.vala) with [Vala Language Server](https://github.com/vala-lang/vala-language-server). If I need to check lint I use [Vala-Lint](https://github.com/vala-lang/vala-lint). Since I clone many projects I made a bash alias for `meson build`. You can add `alias build='meson build --prefix=/usr && cd build'` inro your .bashrc file for quick building.

**Comment by sauntor** [comment]
> I use Visual Studio Code for editing and most times as terminal while developing, sometimes use built-in Terminal app. I use [Vala plugin](https://marketplace.visualstudio.com/items?itemName=prince781.vala) with [Vala Language Server](https://github.com/vala-lang/vala-language-server). If I need to check lint I use [Vala-Lint](https://github.com/vala-lang/vala-lint). Since I clone many projects I made a bash alias for `meson build`. You can add `alias build='meson build --prefix=/usr && cd build'` inro your .bashrc file for quick building.

Thanks a lot!  `Vala plugin` for `VSCode` is great!
